packages/build-info/src/detect-build-system.ts

```diff
@@ -0,0 +1,154 @@
+import { readFileSync } from 'fs'
+import path from 'path'
+
+import { findUpSync } from 'find-up'
+import { PackageJson } from 'read-pkg'
+
+export type BuildSystem = {
+  name: string
+  version?: string | undefined
+}
+
+type BuildSystemHandler = (baseDir: string, rootDir?: string) => Promise<BuildSystem | undefined>
+
+export const detectBuildSystems = async (baseDir: string, rootDir?: string): Promise<BuildSystem[]> => {
+  const buildTools = Object.keys(BUILD_SYSTEMS)
+  const buildSystems = await Promise.all(buildTools.map(async (tool) => await BUILD_SYSTEMS[tool](baseDir, rootDir)))
+
+  return buildSystems.reduce((res, tool) => {
+    if (tool) {
+      res.push(tool)
+    }
+
+    return res
+  }, [] as BuildSystem[])
+}
+
+const BUILD_SYSTEMS: Record<string, BuildSystemHandler> = {
+  nx: async (baseDir: string, rootDir?: string): Promise<BuildSystem | undefined> => {
```

Comment:
This types should not be needed anymore with type inference through the object type

```suggestion
  nx: async (baseDir, rootDir)  => {
```